### PR TITLE
[receiver/purefa] Improvements based on manual testing

### PR DESCRIPTION
**Pull request by dgoscn**
**Description:**
These changes were requested by the Pure Storage team for a better running and clear transparency on the consume from generated data by the receiver.

**Link to tracking Issue:**
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14886

**Testing:**
Same present on **README.md** section

**Documentation:**
Still in progress

### CODE REVIEW

**Review by jpkrohling — APPROVED** [review]

File: receiver/purefareceiver/README.md [hunk 1]
```diff
@@ -25,24 +25,24 @@ extensions:
 receivers:
   purefa:
     endpoint: http://172.0.0.1:9490/metrics
-    arrays:
-      - address: gse-array01
+    array:
+      - address: array01
         auth:
           authenticator: bearertokenauth/array01
     host:
```
Comment (chrroberts-pure) [inline receiver/purefareceiver/README.md:32]: shouldn't this key be 'hosts'?

File: receiver/purefareceiver/testdata/config.yaml [hunk 2]
```diff
@@ -4,24 +4,24 @@ receivers:
 
   purefa/with_custom_intervals:
     endpoint: http://172.31.60.208:9490/metrics
-    arrays:
-      - address: gse-array01
+    array:
+      - address: array01
         auth:
           authenticator: bearertokenauth/array01
     host:
```
Comment (chrroberts-pure) [inline receiver/purefareceiver/testdata/config.yaml:11]: `hosts` should be the key here

File: receiver/purefareceiver/receiver.go [hunk 3]
```diff
@@ -50,19 +50,18 @@ func (r *purefaReceiver) Start(ctx context.Context, compHost component.Host) err
 	scrapeCfgs := []*config.ScrapeConfig{}
 
 	commomLabel := model.LabelSet{
-		"env":           model.LabelValue(r.cfg.Env),
-		"fa_array_name": model.LabelValue(r.cfg.Endpoint),
-		"host":          model.LabelValue(r.cfg.Endpoint),
+		"deployment.environment": model.LabelValue(r.cfg.Env),
+		"host.name":              model.LabelValue(r.cfg.Endpoint),
```
Comment (chrroberts-pure) [inline receiver/purefareceiver/receiver.go:54]: `host.name` should have the same value as `arr.Address` on the endpoints /metrics/array, /metrics/directories, /metrics/pods. 

the label `host` from the metrics from /metrics/hosts, and /metrics/volumes should be re-labeled as `host.name` and should overwrite the `host.name` that comes from the otel-col auto-discovery

File: receiver/purefareceiver/testdata/config.yaml (hunk 2 above)
Comment (dgoscn) [reply receiver/purefareceiver/testdata/config.yaml:11]: For sure, @chrroberts-pure

File: receiver/purefareceiver/receiver.go (hunk 3 above)
Comment (dgoscn) [reply receiver/purefareceiver/receiver.go:54]: Hmm. Yes, I think that I've got. Thanks for this review

cc @jpkrohling

Comment (chrroberts-pure) [reply receiver/purefareceiver/receiver.go:54]: cool. let me build and test, before giving my full approval. Will do first thing tomorrow morning. Thanks!

**Comment by chrroberts-pure** [comment]
factory.go needs to be updates for 'Hosts'

```
@@ -46,7 +46,7 @@ func createDefaultConfig() component.Config {
                Settings: &Settings{
                        ReloadIntervals: &ReloadIntervals{
                                Array:       15 * time.Second,
-                               Hosts:       15 * time.Second,
+                               Host:        15 * time.Second,
                                Directories: 15 * time.Second,
                                Pods:        15 * time.Second,
                                Volumes:     15 * time.Second,
```

File: receiver/purefareceiver/receiver.go [hunk 4]
```diff
@@ -50,19 +50,18 @@ func (r *purefaReceiver) Start(ctx context.Context, compHost component.Host) err
...
-	arrScraper := internal.NewScraper(ctx, internal.ScraperTypeArray, r.cfg.Endpoint, r.cfg.Arrays, r.cfg.Settings.ReloadIntervals.Array, commomLabel)
+	arrScraper := internal.NewScraper(ctx, internal.ScraperTypeArray, r.cfg.Endpoint, r.cfg.Array, r.cfg.Settings.ReloadIntervals.Array, commomLabel)
 	if scCfgs, err := arrScraper.ToPrometheusReceiverConfig(compHost, fact); err == nil {
 		scrapeCfgs = append(scrapeCfgs, scCfgs...)
 	} else {
 		return err
 	}
 
-	hostScraper := internal.NewScraper(ctx, internal.ScraperTypeHost, r.cfg.Endpoint, r.cfg.Hosts, r.cfg.Settings.ReloadIntervals.Host, commomLabel)
+	hostScraper := internal.NewScraper(ctx, internal.ScraperTypeHosts, r.cfg.Endpoint, r.cfg.Hosts, r.cfg.Settings.ReloadIntervals.Host, commomLabel)
```
Comment (chrroberts-pure) [inline receiver/purefareceiver/receiver.go:64]: `r.cfg.Settings.ReloadIntervals.Hosts` to get the new interval imported

**Comment by chrroberts-pure** [comment]
while working through some other issues (ignore the full error here, just pay attention to the labels) it looks like the fa_array_name is being sticky when there are multiple arrays 

```
receivers:
  purefa:
    endpoint: http://localhost:9490/metrics
    hosts:
    - address: array01
      auth:
        authenticator: bearertokenauth/array01
    - address: array05
      auth:
        authenticator: bearertokenauth/array05
```

```
2023-02-10T14:24:10.643-0600	warn	internal/transaction.go:121	Failed to scrape Prometheus endpoint	{"kind": "receiver", "name": "purefa", "data_type": "metrics", "scrape_timestamp": 1676060650482, "target_labels": "{__name__=\"up\", deployment.environment=\"gse-dev\", fa_array_name=\"array05\", host.name=\"http://localhost:9490/metrics\", instance=\"localhost:9490\", job=\"purefa/hosts/array05\"}"}
2023-02-10T14:24:13.331-0600	warn	internal/transaction.go:121	Failed to scrape Prometheus endpoint	{"kind": "receiver", "name": "purefa", "data_type": "metrics", "scrape_timestamp": 1676060653193, "target_labels": "{__name__=\"up\", deployment.environment=\"gse-dev\", fa_array_name=\"array05\", host.name=\"http://localhost:9490/metrics\", instance=\"localhost:9490\", job=\"purefa/hosts/array01\"}"}
```

if it is better to have the configuration like this to avoid the tagging issue above, this could be fine, but needs to be documented.:
```
receivers:
  purefa/array01:
    endpoint: http://localhost:9490/metrics
    hosts:
    - address: array01
      auth:
        authenticator: bearertokenauth/array01
  purefa/array05:
    endpoint: http://localhost:9490/metrics
    hosts:
    - address: array05
      auth:
        authenticator: bearertokenauth/array05
```

File: receiver/purefareceiver/receiver.go (hunk 4 above)
Comment (dgoscn) [reply receiver/purefareceiver/receiver.go:64]: Sure, this make sense

**Comment by dgoscn** [comment]
> factory.go needs to be updates for 'Hosts'
> 
> ```
> @@ -46,7 +46,7 @@ func createDefaultConfig() component.Config {
>                 Settings: &Settings{
>                         ReloadIntervals: &ReloadIntervals{
>                                 Array:       15 * time.Second,
> -                               Hosts:       15 * time.Second,
> +                               Host:        15 * time.Second,
>                                 Directories: 15 * time.Second,
>                                 Pods:        15 * time.Second,
>                                 Volumes:     15 * time.Second,
> ```

@Great appointment, @chrroberts-pure 

> it looks like the fa_array_name is being sticky when there are multiple arrays

I've got it. I will make some tests based on your review. Tomorrow, I will send de PR with the properly changes.

File: receiver/purefareceiver/config.go [hunk 5]
```diff
@@ -57,7 +57,7 @@ type Settings struct {
 
 type ReloadIntervals struct {
 	Array       time.Duration `mapstructure:"array"`
-	Host        time.Duration `mapstructure:"host"`
+	Hosts       time.Duration `mapstructure:"host"`
```
Comment (codeboten) [inline receiver/purefareceiver/config.go:60]: Why is this plural here but the config definition is singular?

**Review by codeboten — COMMENTED** [review]
Please add a changelog to notify the users of the changes in this PR.

**Comment by dgoscn** [comment]
> Please add a changelog to notify the users of the changes in this PR.

Yes, sure @codeboten! Thanks for the review

File: receiver/purefareceiver/config.go (hunk 5 above)
Comment (chrroberts-pure) [reply receiver/purefareceiver/config.go:60]: Agree. The OpenMetrics endpoint for the [PureFA OpenMetrics Exporter](https://github.com/PureStorage-OpenConnect/pure-fa-openmetrics-exporter#scraping-endpoints) is /metrics/hosts and needs to be reflected in the same plurality.

Comment (chrroberts-pure) [inline receiver/purefareceiver/config.go:60]:
```suggestion
	Hosts       time.Duration `mapstructure:"hosts"`
```

File: receiver/purefareceiver/config.go [hunk 6]
```diff
@@ -32,8 +32,8 @@ type Config struct {
 	// Settings contains settings for the individual scrapers
 	Settings *Settings `mapstructure:"settings"`
 
-	// Arrays represents the list of arrays to query
-	Arrays []internal.ScraperConfig `mapstructure:"arrays"`
+	// Array represents the list of arrays to query
```
Comment (codeboten) [inline receiver/purefareceiver/config.go:35]: This comment makes me think array should remain plurals, as its a list of arrays. if i read correctly, the endpoint is mapped to `/array`, but as a user, i wouldn't know which of these configuration options need to be plural vs singular if some singular and plural need lists as configurations.

**Review by codeboten — COMMENTED** [review]
Thanks for addressing the changelog comment, i have one more comment/question

File: receiver/purefareceiver/config.go (hunk 6 above)
Comment (dgoscn) [reply receiver/purefareceiver/config.go:35]: Excellent review @codeboten I think that was surgical. But the comment that Christopher just sent, resolves this ambiguity for the config.

**Review by codeboten — APPROVED** [review]

**Pull request merged**